### ISSUE TITLE
How to change to PEFT model dynamically?

### ISSUE THREAD

**Issue by whr819987540** [huggingface/peft#1829]
python==3.7.12
PEFT==0.3.0

@BenjaminBossan 

I fine-tune the eleventh transformer of Bert as below:

```bash
target_modules = []
target_modules.append("11.attention.self.query")
target_modules.append("11.attention.self.value")

lora_config = LoraConfig(
    r = self.args.lora_rank,
    lora_alpha = self.args.lora_alpha,
    target_modules = target_modules,
    lora_dropout = 0.05,
    bias = "none"
)
```

After training for a few epochs, I also want to fine-tune the first transformer. How to achieve this?

**Comment by BenjaminBossan** [comment]
This is not directly possible. What you could try is to add all the layers you want to eventually train into `target_modules`. Then, you go through the modules and manually disable the gradient of those you _don't_ want to train:

```python
target_modules = [
    "0.attention.self.query", "0.attention.self.value",
    "11.attention.self.query", "11.attention.self.value",
]
...
model = get_peft_model(...)
for name, module in model.named_modules():
    if name ...:
        module.requires_grad_(False)
```

I've never tried this, so not 100% sure if it'll work, but worth a try.

**Comment by BenjaminBossan** [comment]
Yes, if you create more LoRA layers, the random seed will be ticked more often. But I don't understand what the big issue would be with that. Maybe you could show your code and explain what you expect vs what actually happens.

**Comment by whr819987540** [comment]
Just to know whether it's the reason of random seed. 

Thanks a lot for your reply.
